File: backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/suggestion/dto/ListByUsernameDto.java
package pt.ulisboa.tecnico.socialsoftware.tutor.suggestion.dto;

import java.io.Serializable;
import java.util.List;

public class ListByUsernameDto implements Serializable {

    public List<SuggestionDto> _suggslist;
    private Integer numberofsuggs = 0;
    private Integer numberofapprovedsuugs = 0;


    public ListByUsernameDto(){
    }

    public void setNumberofsuugs(Integer getnumberofsuggs) {
    }

    public List<SuggestionDto> getListByUsernameDto () {return this._suggslist;}

    public void setListByUsernameDto(List<SuggestionDto> suggs) {this._suggslist = suggs;}


    public Integer getNumberofapprovedsuugs() {
        return numberofapprovedsuugs;
    }

    public void setNumberofapprovedsuugs(Integer numberofapprovedsuugs) {
        this.numberofapprovedsuugs = numberofapprovedsuugs;
    }

    public Integer getNumberofsuggs() {
        return numberofsuggs;
    }

    public void setNumberofsuggs(Integer numberofsuggs) {
        this.numberofsuggs = numberofsuggs;
    }
}

File: frontend/src/models/management/ListByUsernameDto.ts
import Suggestion from '@/models/management/Suggestion';

export default class ListByUsernameDto {
  _suggslist!: Suggestion[];
  numberofsuggs!: number;
  numberofapprovedsuugs!: number;

  constructor(jsonObj?: ListByUsernameDto) {
    if (jsonObj) {
      this.numberofsuggs = jsonObj.numberofsuggs;
      this.numberofapprovedsuugs = jsonObj.numberofapprovedsuugs;

      if (jsonObj._suggslist != null)
        this._suggslist = jsonObj._suggslist.map(
          (suggestion: Suggestion) => new Suggestion(suggestion)
        );
    }
  }
}
